runtime/appruntime/apisdk/service/service.go

```diff
@@ -70,7 +71,21 @@ func doSetupService[T any](mgr *Manager, decl *Decl[T]) (err error) {
 		setupFn = func() (*T, error) { return new(T), nil }
 	}
 
-	i, err := setupFn()
+	var i *T
+
+	// backoff retry for service initialization
+	//
+	// this is to handle the case where a service is dependent on another service during a cold start
+	// of the application, and the dependent service is not yet ready
+	for attempt := 0; attempt < 5; attempt++ {
```

Comment:
I don't think we should do this; we haven't documented or required that the initialization function be idempotent or designed to handle being retried within the same process

Reply:
I don’t agree for now. The reason is this change broke the test app without; and most things in there are because customers have also got that behaviour. In which case we either need to document a breaking change and give people time to migrate - or support this and document a new retry method. 

We could maybe only retry on connection failed errors; so the retry only applies to errors coming from our own external calls

Reply:
Let's discuss on Monday, but this is mainly a concern when developing locally when all services spin up at the same time and there's only one instance of them. A proxy that waits for the service to start up would solve this.

Reply:
This wouldn’t solve a cold start in the cloud though; sign up a new environment of an existing app with this behaviour. 

There’s no proxy waiting to delay the requests there.

Reply:
Relying on another service being up to start up is always going to be error-prone. But you can implement such retries yourself if it's appropriate for your use case.